lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -0,0 +1,201 @@
+selection cuts and/or IRF FITS files
+
+Simple usage with argument aliases and default config file for selection cuts:
+
+lstchain_create_irf_files
+    --d /path/to/DL2_data_file.h5
+    --o /path/to/DL3/file/
+    --add_irf True
+    --irf /path/to/irf.fits.gz
+    --source_name Crab
+"""
+
+import os
+from astropy.io import fits
+import astropy.units as u
+
+from ctapipe.core import Tool, traits, Provenance, ToolConfigurationError
+from lstchain.io import read_data_dl2_to_QTable, read_configuration_file
+from lstchain.reco.utils import filter_events, get_effective_time
+from lstchain.paths import run_info_from_filename, dl2_to_dl3_filename
+from lstchain.irf import create_event_list
+
+from pyirf.utils import calculate_source_fov_offset
+
+__all__ = ["DataReductionFITSWriter"]
+
+
+class DataReductionFITSWriter(Tool):
+    name = "DataReductionFITSWriter"
+    description = __doc__
+
+    input_dl2 = traits.Path(
+        help="Input data DL2 file", exists=True, directory_ok=False, file_ok=True
+    ).tag(config=True)
+
+    output_dl3_path = traits.Path(
+        help="DL3 output filedir", directory_ok=True, file_ok=False
+    ).tag(config=True)
+
+    add_irf = traits.Bool(
+        help="True for adding IRF fits file to the DL3 file",
+        default_value=True,
+    ).tag(config=True)
+
+    input_irf = traits.Path(
+        help="Compressed FITS file of IRFs",
+        exists=True,
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    source_name = traits.Unicode(help="Name of Source").tag(config=True)
+
+    config_file = traits.Path(
+        help="Config file for selection cuts",
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    overwrite = traits.Bool(
+        help="If True, overwrites existing output file without asking",
+        default_value=True,
+    ).tag(config=True)
+
+    provenance_log = traits.Path(
+        help="Path for the Provenance log",
+        directory_ok=False,
+    ).tag(config=True)
+
+    aliases = {
+        "input_dl2": "DataReductionFITSWriter.input_dl2",
+        "d": "DataReductionFITSWriter.input_dl2",
+        "output_dl3_path": "DataReductionFITSWriter.output_dl3_path",
+        "o": "DataReductionFITSWriter.output_dl3_path",
+        "add_irf": "DataReductionFITSWriter.add_irf",
+        "input_irf": "DataReductionFITSWriter.input_irf",
+        "irf": "DataReductionFITSWriter.input_irf",
+        "config_file": "DataReductionFITSWriter.config_file",
+        "conf": "DataReductionFITSWriter.config_file",
+        "source_name": "DataReductionFITSWriter.source_name",
+        "provenance_log": "DataReductionFITSWriter.provenance_log",
+        "prov": "DataReductionFITSWriter.provenance_log",
+    }
+
+    flags = {
+        "overwrite": (
+            {"DataReductionFITSWriter": {"overwrite": True}},
+            "overwrite output file",
+        )
+    }
+
+    def __init__(self, **kwargs):
```

Comment:
Why set all these variables to None initially?

Reply:
I think somehow I assumed it was necessary. I will remove them from other tools as well.